android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -299,12 +299,14 @@ public String getInstallReferrer() {
     constants.put("uniqueId", Settings.Secure.getString(this.reactContext.getContentResolver(), Settings.Secure.ANDROID_ID));
     constants.put("systemManufacturer", Build.MANUFACTURER);
     constants.put("bundleId", packageName);
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
-      try {
+    try {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
         constants.put("userAgent", WebSettings.getDefaultUserAgent(this.reactContext));
-      } catch (RuntimeException e) {
+      } else {
         constants.put("userAgent", System.getProperty("http.agent"));
       }
+    } catch (RuntimeException e) {
+      constants.put("userAgent", System.getProperty("http.agent"));
```

Comment:
Hmm - from what I read (https://stackoverflow.com/a/5261472/9910298) it looks like the http.agent property is the device user agent and will be different than the WebView user agent, so they should not be conflated, but we already did so with the current implementation. Given that, this seems fine as a strategy.